src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -33,8 +33,17 @@ public AuthorizationMiddleware(RequestDelegate next, IAuthorizationPolicyProvide
     {
         _next = next ?? throw new ArgumentNullException(nameof(next));
         _policyProvider = policyProvider ?? throw new ArgumentNullException(nameof(policyProvider));
+
+        // Only try to cache combined policies for the default policy provider
+#if NETCOREAPP
+        CacheCombinedPolicy = _policyProvider.CanCachePolicy;
+#else
+        CacheCombinedPolicy = _policyProvider.GetType() == typeof(DefaultAuthorizationPolicyProvider);
+#endif
```

Comment:
The middleware is netcoreapp only. We don't need this check.

Reply:
Ah nice, that lets me get rid of the property entirely now